src/ui/wxWidgets/AddEditPropSheetDlg.cpp

```diff
@@ -1347,52 +1343,95 @@ static struct {short pv; wxString name;}
   }
 }
 
-void AddEditPropSheetDlg::UpdateExpTimes()
+// Called once to initialize the expiration controls
+void AddEditPropSheetDlg::InitializeExpTimes()
 {
   // From m_item to display
-
   m_Item.GetXTime(m_tttExpirationTime);
   m_Item.GetXTimeInt(m_ExpirationTimeInterval);
-  m_ExpirationTime = m_CurrentExpirationTime = m_Item.GetXTimeL().c_str();
+  m_OriginalDayttt = 0;
+  m_OriginalExpirationDate = m_Item.GetXTimeL().c_str();
 
-  if (m_ExpirationTime.empty())
-    m_ExpirationTime = m_CurrentExpirationTime = _("Never");
+  // Special case: Some entries, created with recent versions of pwsafe, might have
+  // an interval but no date, which is interpreted as "Never".  We are going to ignore
+  // it and use the user-set default.  If an expiry change is made, the entry will
+  // be re-written correctly.
+  int defInterval = PWSprefs::GetInstance()->GetPref(PWSprefs::DefaultExpiryDays);
 
-  wxCommandEvent dummy;
   wxDateTime exp;
```

Comment:
```suggestion
  wxDateTime expiryDate;
```
(I had to scroll back to this line a few times to remind myself what this variable contains)

Reply:
Done